packages/web/components/swap-tool/alt.tsx

```diff
@@ -0,0 +1,694 @@
+import { isNil } from "@osmosis-labs/utils";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import { parseAsBoolean, useQueryState } from "nuqs";
+import {
+  FunctionComponent,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { useMeasure } from "react-use";
+
+import { Icon } from "~/components/assets";
+import {
+  AssetFieldset,
+  AssetFieldsetFooter,
+  AssetFieldsetHeader,
+  AssetFieldsetHeaderBalance,
+  AssetFieldsetHeaderLabel,
+  AssetFieldsetInput,
+  AssetFieldsetTokenSelector,
+} from "~/components/complex/asset-fieldset";
+import { tError } from "~/components/localization";
+import { TradeDetails } from "~/components/swap-tool/trade-details";
+import { Button } from "~/components/ui/button";
+import { EventName, EventPage } from "~/config";
+import {
+  useAmplitudeAnalytics,
+  useDisclosure,
+  useFeatureFlags,
+  useOneClickTradingSession,
+  useSlippageConfig,
+  useTranslation,
+  useWalletSelect,
+  useWindowSize,
+} from "~/hooks";
+import { useSwap } from "~/hooks/use-swap";
+import { useGlobalIs1CTIntroModalScreen } from "~/modals";
+import { AddFundsModal } from "~/modals/add-funds";
+import { ReviewOrder } from "~/modals/review-order";
+import { TokenSelectModalLimit } from "~/modals/token-select-modal-limit";
+import { useStore } from "~/stores";
+import { formatPretty, getPriceExtendedFormatOptions } from "~/utils/formatter";
+
+export interface SwapToolProps {
+  fixedWidth?: boolean;
+  useOtherCurrencies: boolean | undefined;
+  useQueryParams: boolean | undefined;
+  onRequestModalClose?: () => void;
+  swapButton?: React.ReactElement;
+  initialSendTokenDenom?: string;
+  initialOutTokenDenom?: string;
+  page: EventPage;
+  forceSwapInPoolId?: string;
+  onSwapSuccess?: (params: {
+    sendTokenDenom: string;
+    outTokenDenom: string;
+  }) => void;
+}
+
+export const AltSwapTool: FunctionComponent<SwapToolProps> = observer(
+  ({
+    useOtherCurrencies,
+    useQueryParams,
+    onRequestModalClose,
+    swapButton,
+    initialSendTokenDenom,
+    initialOutTokenDenom,
+    page,
+    forceSwapInPoolId,
+    onSwapSuccess,
+  }) => {
+    const { chainStore, accountStore } = useStore();
+    const { t } = useTranslation();
+    const { chainId } = chainStore.osmosis;
+    const { isMobile } = useWindowSize();
+    const { logEvent } = useAmplitudeAnalytics();
+    const { isLoading: isWalletLoading, onOpenWalletSelect } =
+      useWalletSelect();
+    const featureFlags = useFeatureFlags();
+    const [, setIs1CTIntroModalScreen] = useGlobalIs1CTIntroModalScreen();
+    const { isOneClickTradingEnabled } = useOneClickTradingSession();
+    const [isSendingTx, setIsSendingTx] = useState(false);
+
+    const [_, setType] = useQueryState("type");
+
+    useEffect(() => {
+      setType(null);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
Nit: If we want this to only occur on mount, we can use the `useMount` hook to make this more explicit.